packages/astro/src/events/error.ts

```diff
@@ -64,12 +65,51 @@ export function eventError({
 	cmd: string;
 	isFatal: boolean;
 }): { eventName: string; payload: ErrorEventPayload }[] {
+	const errorData =
+		AstroError.is(err) && err.errorCode ? getErrorDataByCode(err.errorCode)?.data : undefined;
+
 	const payload: ErrorEventPayload = {
-		code: err.code || AstroErrorData.UnknownError.code,
+		code: err.code || err.errorCode || AstroErrorData.UnknownError.code,
+		name: err.name,
 		plugin: err.plugin,
 		cliCommand: cmd,
 		isFatal: isFatal,
-		anonymousMessageHint: anonymizeErrorMessage(err.message),
+		anonymousMessageHint:
+			errorData && errorData.message
+				? getSafeErrorMessage(errorData.message)
+				: anonymizeErrorMessage(err.message),
 	};
 	return [{ eventName: EVENT_ERROR, payload }];
 }
+
+/**
+ * Safely get the error message from an error, even if it's a function.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+function getSafeErrorMessage(message: string | Function): string {
+	if (typeof message === 'string') {
+		return message;
+	} else {
+		return String.raw({
+			raw: extractStringFromFunction(message.toString()),
+		});
+	}
+
+	function extractStringFromFunction(func: string) {
```

Comment:
Im not sure about this one... from reading it and please correct me, but this extracts the name from the function, 🤔
but this only works with the use case of anonymous functions, what about named functions? or is it outwith the scope of the method?
Also killer ninja regexing 🐱‍👤

Reply:
Yes! Named functions are outside the scope of this function, in our error definitions, we only use arrow functions (named functions would be totally unnecessary, as we'd never refer to them by name). The functions there are also on purpose simple, as they're also used for documentation